Docs/source/developers/testing.rst

```diff
@@ -29,24 +23,52 @@ Local tests every night
+    ./run_test.sh pml_x_yee pml_x_ckc pml_x_psatd
 
-* ``export WARPX_TEST_DIM=3``, ``export WARPX_TEST_DIM=2`` or ``export WARPX_TEST_DIM=RZ`` in order to select only the tests that correspond to this dimensionality
-* ``export WARPX_TEST_ARCH=CPU`` or ``export WARPX_TEST_ARCH=GPU`` in order to run the tests on CPU or GPU respectively.
+Note that the script `./run_test.sh <https://github.com/ECP-WarpX/WarpX/blob/development/run_test.sh>`__ runs the tests with the exact same compile-time options and runtime options used to run the tests remotely.
 
-The command above (without command line arguments) runs all the tests defined in [Regression/WarpX-tests.ini](./Regression/WarpX-tests.ini). In order to run single tests, pass the test names as command line arguments:
+Moreover, the script `./run_test.sh <https://github.com/ECP-WarpX/WarpX/blob/development/run_test.sh>`__ compiles all the executables that are necessary in order to run the chosen tests.
+The default number of threads allotted for compiling, initially set as ``numMakeJobs = 8`` in `./Regression/WarpX-tests.ini <https://github.com/ECP-WarpX/WarpX/blob/ad74bcbdd131a8797339ba38370b1195d0aecffb/Regression/WarpX-tests.ini#L20>`__, is overwritten as ``numMakeJobs = 2`` in `./Regression/prepare_file_travis.py <https://github.com/ECP-WarpX/WarpX/blob/ad74bcbdd131a8797339ba38370b1195d0aecffb/Regression/prepare_file_travis.py#L74>`__, in order to avoid overloading the remote host.
+However, when running the tests on a local machine, it is usually possible and convenient to allot more threads for compiling, in order to speed up the builds.
+This can be accomplished by modifying locally the assignment ``numMakeJobs = 2`` in `./Regression/prepare_file_travis.py <https://github.com/ECP-WarpX/WarpX/blob/ad74bcbdd131a8797339ba38370b1195d0aecffb/Regression/prepare_file_travis.py#L74>`__, with the preferred number of threads that fits your local machine.
+Note that this will not change the number of threads used to run each test, but only the number of threads used to compile each executable necessary to run the tests.
+
+Once the execution of `./run_test.sh <https://github.com/ECP-WarpX/WarpX/blob/development/run_test.sh>`__ is completed, you can find all the relevant files associated with each test in one single directory.
+For example, if you run the single test ``pml_x_yee``, as shown above, on 04/30/2021,  you can find all relevant files in the directory ``./test_dir/rt-WarpX/WarpX-tests/2021-04-30/pml_x_yee/``.
+The content of this directory will look like the following (possibly including backtraces if the test crashed at runtime):
 
 .. code-block:: sh
 
-    ./run_test.sh test1 test2
-    # For instance
-    ./run_test.sh PlasmaAccelerationBoost3d Larmor
+    $ ls ./test_dir/rt-WarpX/WarpX-tests/2021-04-30/pml_x_yee/
+    analysis_pml_yee.py*    # Python analysis script
+    inputs_2d               # input file
+    main2d.gnu.TEST.TPROF.MTMPI.OMP.QED.GPUCLOCK.ex*  # executable
```

Comment:
Yes, that is just how my Ubuntu terminal marks executables, so I simply reported it as it is in this example.

And I agree that we should probably rename `prepare_file_travis.py` since we no longer depend on Travis, indeed (as well as clean up some comments that reference Travis and might have been left over here and there).

Reply:
yep, let's remove the `*` here as it is a bit confusing and non-standard

Reply:
Sounds good, done it in https://github.com/ECP-WarpX/WarpX/pull/1926/commits/262a605e713bd62b3d0b153ba5e35269284a80fd.